onnxruntime/core/providers/coreml/model/model.mm

```diff
@@ -244,13 +244,20 @@ - (void)dealloc {
                    1,
                    std::multiplies<int64_t>());
 
-    if (output_tensor.tensor_info.data_type != ONNX_NAMESPACE::TensorProto_DataType_FLOAT) {
-      return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL,
-                             "Input data type is not float, actual type: ",
+    size_t output_data_byte_size = 0;
+    auto type = output_tensor.tensor_info.data_type;
+    switch ( type ) {
+      case ONNX_NAMESPACE::TensorProto_DataType_FLOAT:
+        output_data_byte_size = num_elements * sizeof(float);
+        break;
+      case ONNX_NAMESPACE::TensorProto_DataType_INT32:
+        output_data_byte_size = num_elements * sizeof(int32_t);
+        break;
+      default:
+        return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL,
+                             "Output data type is not float/int32, actual type: ",
                              output_tensor.tensor_info.data_type);
```

Comment:
>output_tensor.tensor_info.data_type [](http://example.com/codeflow?start=29&length=35)

nit: since we introduced `type`, that could be used here too. `type` could be const